src/_ert/forward_model_runner/client.py

```diff
@@ -26,107 +22,139 @@ class ClientConnectionClosedOK(Exception):
 
 
 class Client:
-    DEFAULT_MAX_RETRIES = 10
-    DEFAULT_TIMEOUT_MULTIPLIER = 5
-    CONNECTION_TIMEOUT = 60
+    DEFAULT_MAX_RETRIES = 5
+    DEFAULT_ACK_TIMEOUT = 5
+    _receiver_task: asyncio.Task[None] | None
 
     def __enter__(self) -> Self:
+        self.loop.run_until_complete(self.__aenter__())
         return self
 
+    def term(self) -> None:
+        self.socket.close()
+        self.context.term()
+
     def __exit__(self, exc_type: Any, exc_value: Any, exc_traceback: Any) -> None:
-        if self.websocket is not None:
-            self.loop.run_until_complete(self.websocket.close())
+        self.loop.run_until_complete(self.__aexit__(exc_type, exc_value, exc_traceback))
         self.loop.close()
 
-    async def __aenter__(self) -> "Client":
+    async def __aenter__(self) -> Self:
+        await self.connect()
         return self
 
     async def __aexit__(
         self, exc_type: Any, exc_value: Any, exc_traceback: Any
     ) -> None:
-        if self.websocket is not None:
-            await self.websocket.close()
+        try:
+            await self._send("DISCONNECT")
+        except ClientConnectionError:
+            logger.error("No ack for dealer disconnection. Connection is down!")
+        finally:
+            self.socket.disconnect(self.url)
+            await self._term_receiver_task()
+            self.term()
+
+    async def _term_receiver_task(self) -> None:
+        if self._receiver_task and not self._receiver_task.done():
+            self._receiver_task.cancel()
+            await asyncio.gather(self._receiver_task, return_exceptions=True)
+            self._receiver_task = None
 
     def __init__(
         self,
         url: str,
         token: str | None = None,
-        cert: str | bytes | None = None,
-        max_retries: int | None = None,
-        timeout_multiplier: int | None = None,
+        dealer_name: str | None = None,
+        ack_timeout: float | None = None,
     ) -> None:
-        if max_retries is None:
-            max_retries = self.DEFAULT_MAX_RETRIES
-        if timeout_multiplier is None:
-            timeout_multiplier = self.DEFAULT_TIMEOUT_MULTIPLIER
-        if url is None:
-            raise ValueError("url was None")
+        self._ack_timeout = ack_timeout or self.DEFAULT_ACK_TIMEOUT
         self.url = url
         self.token = token
-        self._additional_headers = Headers()
+
+        # Set up ZeroMQ context and socke
+        self._ack_event: asyncio.Event = asyncio.Event()
+        self.context = zmq.asyncio.Context()
+        self.socket = self.context.socket(zmq.DEALER)
+        self.socket.setsockopt(zmq.LINGER, 0)
```

Comment:
What does this option do?

Reply:
Hm, maybe I can put a comment there. It is to prevent queuing messages and instead they are sent right away.